inspirehep/modules/recordvalidators/validators/rules.py

```diff
@@ -0,0 +1,402 @@
+    return {}
+
+
+def check_cnum_if_doctype_value_conference_paper_present(record):
+    doctype_values = record.get('document_type', [])
+    if 'conference paper' in doctype_values:
+        if not check_if_field_exists_in_dict_list_x_times('cnum', record.get('publication_info', [])):
+            raise ValidationError(key='globalErrors',
+                                  message=FIELD_VALUE_REQUIRE_FIELD_TEMPLATE.format(
+                                      field='document_type',
+                                      value='conference paper',
+                                      required='cnum'),
+                                  type='Warning')
+    return {}
+
+
+def check_if_2_cnum_exist_in_publication_info(record):
+    if check_if_field_exists_in_dict_list_x_times('cnum', record.get('publication_info', []), limit=2):
+        raise ValidationError(key='globalErrors',
+                              message="2 cnums found in 'publication info' field.",
+                              type='Warning')
+
+
+def check_doctype_values_if_cnum_present(record):
+    if check_if_field_exists_in_dict_list_x_times('cnum', record.get('publication_info', [])):
+        required_values = ['proceedings', 'conference_paper']
+        if check_field_values_not_in_required_values_for_record(record, 'document_type', required_values):
+            raise ValidationError(key='globalErrors',
+                                  message=FIELD_REQUIRE_FIELD_VALUES_TEMPLATE.format(
+                                      field='cnum',
+                                      required='document_type',
+                                      values=required_values),
+                                  type='Warning')
+    return {}
+
+
+def check_accelerator_experiments_if_collaborations_exist(record):
+    if 'collaborations' in record:
+        if not 'accelerator_experiments' in record:
+            raise ValidationError(key='globalErrors',
+                                  message=FIELD_REQUIRE_FIELD_TEMPLATE.format(
+                                      field='collaborations',
+                                      required='accelerator_experiments'),
+                                  type='Warning')
+    return {}
+
+
+def check_if_reportnumber_exist_in_other_records(record):
+    try:
+        check_if_listfield_property_values_are_valid(record=record,
+                                                     field='report_numbers',
+                                                     prop='value',
+                                                     checker=query_db_for_duplicates_for_field_listitem_with_index,
+                                                     errortype='Warning')
+    except ValidationError:
+        raise
+    return {}
+
+
+def check_external_urls_if_work(record):
+    def _check_if_url_works(field, prop, value, index, errortype):
+        try:
+            resp = requests.get(value)
+            if not resp.ok:
```

Comment:
You should be able to get more relevant error from the status code and maybe there is a message in the response. To see if there is 404, 500, 3xx,..